packages/playwright/src/util.ts

```diff
@@ -320,13 +320,16 @@ export function resolveImportSpecifierExtension(resolved: string): string | unde
   }
 
   if (dirExists(resolved)) {
+    if (dontResolveDirectories)
```

Comment:
Let's make this check before doing an expensive `dirExists()` call.

Reply:
done in https://github.com/microsoft/playwright/pull/31915/commits/8fe370acf8e4e802f00e640c7ba0eb62069bf5e4